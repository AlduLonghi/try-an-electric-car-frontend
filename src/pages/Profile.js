import { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import fetchConfig from '../helpers/fetch';
import Navbar from '../components/Navbar';
import '../styles/profile.scss';
import baseUrl from '../helpers/base-url';
import LoadingWheel from '../components/LoadingWheel';

const Profile = ({ user }) => {
  const [appointments, setAppointments] = useState();

  useEffect(() => {
    fetch(`${baseUrl}/appointments`, fetchConfig())
      .then(res => {
        if (res.ok) {
          res.json().then(jsonRes => {
            setAppointments(jsonRes.data);
          });
        }
      });
  }, []);

  const handleOnClick = id => {
    fetch(`${baseUrl}/appointments/${id}`, {
      ...fetchConfig(),
      method: 'DELETE',
    }).then(res => {
      if (res.ok) {
        setAppointments(appointments.filter(app => app.id !== id));
      }
    });
  };

  let toRenderComponent;
  if (appointments) {
    toRenderComponent = (
      <div>
        {appointments.map(app => {
          const date = new Date(app.datetime);
          return (
            <div key={app.id} className="card mx-auto mt-2">
              <div className="card-body">
                <h5 className="card-title font-weight-bold">{app.model}</h5>
                <p className="card-text">{date.toString()}</p>
                <button type="button" onClick={() => handleOnClick(app.id)} className="btn btn-danger ml-auto">Cancel</button>
              </div>
            </div>
          );
        })}
      </div>
    );
  } else {
    toRenderComponent = (
      <div className="w-100 h-100">
        <LoadingWheel />
      </div>
    );
  }

  return (
    <div className="h-100 profile-cont">
      <Navbar />
      <main className="h-100">
        <div className="text-center models-text-cont">
          <h1 className="font-weight-bold">{`Hello, ${user.name}`}</h1>
          <p className="select-a-car-text font-weight-bold mt-2">These are your test drive appointments</p>
          <span className="line-span d-block mx-auto mt-5" />
        </div>
        {toRenderComponent}
      </main>
    </div>
  );
};

const mapStateToProps = state => ({
  user: state.user.user,
});

Profile.propTypes = {
  user: PropTypes.instanceOf(Object).isRequired,
};

export default connect(mapStateToProps)(Profile);
