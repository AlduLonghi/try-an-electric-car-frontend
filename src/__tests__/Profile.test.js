import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Profile from '../pages/Profile';
import store from '../redux/store';
import { MemoryRouter } from 'react-router';

it('renders Login page correctly', () => {
    const tree = renderer.create(
        <Provider store={store}>
          <MemoryRouter initialEntries={['/profile']}>
            <Profile />
          </MemoryRouter>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});