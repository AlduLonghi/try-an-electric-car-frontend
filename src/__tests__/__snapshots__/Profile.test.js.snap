// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Login page correctly 1`] = `
<div
  className="h-100 profile-cont"
>
  <div>
    <div>
      <div
        className="collapse-btn"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          <span
            className="material-icons"
          >
            drag_handle
          </span>
        </button>
        <button
          className="logout-btn text-center display-block"
          onClick={[Function]}
          type="button"
        >
          Logout
        </button>
      </div>
    </div>
    <nav
      className="display-none"
    >
      <h1
        className="text-center"
      >
        Electricar
      </h1>
      <ul
        className="text-uppercase font-weight-bold py-1 text-center d-flex flex-column justify-content-center"
      >
        <li>
          <a
            aria-current={null}
            className="a-navbar"
            href="/lifestyle"
            onClick={[Function]}
          >
            lifestyle
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            className="a-navbar"
            href="/models"
            onClick={[Function]}
          >
            models
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            className="a-navbar active-navlink"
            href="/profile"
            onClick={[Function]}
            style={Object {}}
          >
            profile
          </a>
        </li>
      </ul>
      <button
        className="logout-btn text-center display-block"
        onClick={[Function]}
        type="button"
      >
        Logout
      </button>
      <footer
        className="as-navbar-component"
      >
        <div
          className="d-flex justify-content-around w-50 mx-auto mb-2"
        >
          <a
            href="https://github.com/AlduLonghi"
          >
            <img
              alt="github"
              src="github.svg"
            />
          </a>
          <a
            href="https://www.linkedin.com/in/aldanalonghi/"
          >
            <img
              alt="linkedin"
              src="linkedin.svg"
            />
          </a>
          <a
            href="https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign"
          >
            <img
              alt="behance"
              src="behance.svg"
            />
          </a>
        </div>
        <p
          className="text-center p-footer"
        >
          Â©2021 Aldana Longhi
        </p>
      </footer>
    </nav>
  </div>
  <main
    className="h-100"
  >
    <div
      className="text-center models-text-cont"
    >
      <h1
        className="font-weight-bold"
      >
        Hello, undefined
      </h1>
      <p
        className="select-a-car-text font-weight-bold mt-2"
      >
        These are your test drive appointments
      </p>
      <span
        className="line-span d-block mx-auto mt-5"
      />
    </div>
    <div
      className="w-100 h-100"
    >
      <div
        className="loading loading4"
      />
    </div>
  </main>
</div>
`;
